doc/specs/#658 - WinGet Download.md

```diff
@@ -0,0 +1,209 @@
+---
+author: Roy MacLachlan RDMaclachlan/roy.maclachlan@microsoft.com
+created on: 2023-02-09
+last updated: 2023-02-09
+issue id: 658
+---
+
+# Spec Title
+
+"For [#658](https://github.com/microsoft/winget-cli/issues/658)"
+
+## Abstract
+
+This spec describes the functionality and high-level implementation design for the downloading of applications using the Windows Package Manager.
+
+## Inspiration
+
+This is inspired by customer feedback, and a need for broader application deployments:
+* Customers want to share the installer with an offline device. 
+
+## Solution Design
+
+The Windows Package Manager will implement a new command `WinGet Download` that will offer customers with the ability to download a single package's installers. The new command will make use of the following optional parameter values:
+* `--id <Identifier>`
+* `--name <name>`
+* `--moniker <moniker>`
+* `--scope <scope>`
+* `--locale <locale>`
+* `--fileName <name>`
+* `-v, --version <version>`
+* `-s, --source <Source>`
+* `-a, --architecture <architecture>`
+* `-e, --exact`
+* `-o, --output <path>`
+* `-t, --type <installer type>`
+* `--ignore-security-hash`
+* `--accept-package-agreements`
+* `--help`
+* `--wait`
+* `--verbose, --verbose-logs`
+
+An update to the existing schema is required for the inclusion of the `<boolean> isDownloadable` property. If this property is set to `True` or is not specified, then the package's installer will be considered downloadable. If the property is set to `False`, then the request to download the package's installer will be refused.
+
+Downloading the package's installer will still require that the package's installer hash be verified before becoming available to the user to interact with. The installer will be downloaded to the device with a `*.tmp` file extension that will be removed after the installer's hash has been validated. If the hash validation fails, the package's installer will be removed from the device.
```

Comment:
nit: we already have temp file mechanisms, it's not "*.tmp", it's a guid at the moment

Reply:
I am ok with that. Will update the spec.